agent/xds/envoy_versioning.go

```diff
@@ -0,0 +1,84 @@
+package xds
+
+import (
+	"fmt"
+	"regexp"
+
+	envoy "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoycore "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/hashicorp/go-version"
+)
+
+var (
+	// minSafeRegexVersion reflects the minimum version where we could use safe_regex instead of regex
+	//
+	// NOTE: the first version that no longer supported the old style was 1.13.0
+	minSafeRegexVersion = version.Must(version.NewVersion("1.11.2"))
```

Comment:
Conveniently for 1.8.x all of the [officially supported versions](https://www.consul.io/docs/connect/proxies/envoy#supported-versions) are covered by this pattern.

Reply:
I could have elected to avoid doing any sniffing at all here, but upcoming work would require us to add it right back in again anyway.

Reply:
One thing we could start doing after this lands is to put in guardrails when an older or newer envoy dials in to a consul agent. Maybe not going so far as to reject the connection, but we could start logging a "this envoy is unsupported" warning message for stuff outside of our support matrix.